### PR TITLE
do some auto focus; bump 0.2.2

### PR DESCRIPTION
**Pull request by tiye**
Doing some autofocus in code so that we get more fluent keyboard usages. Previews at http://repo.respo.site/alerts/

### CODE REVIEW

File: src/respo_alerts/comp/alerts.cljs [hunk 1]
```diff
@@ -19,7 +20,14 @@
...
    (span
-    {:style {:cursor :pointer}, :on-click (fn [e d! m!] (m! (assoc state :show? true)))}
+    {:style {:cursor :pointer},
+     :on-click (fn [e d! m!]
+       (m! (assoc state :show? true))
+       (js/setTimeout
+        (fn []
+          (let [target (.querySelector js/document (str "." schema/confirm-button-name))]
+            (if (some? target) (.focus target))))
+        50))}
```
Comment (lilydjwg) [inline src/respo_alerts/comp/alerts.cljs:30]: I have several questions:

* Is there a better way than `setTimeout`?
* Why not `getElementsByClassName`?
* Is it possible that there are multiple elements with the class?
* What if the caller wants the cancel button to be default? (e.g. "Destroy the world?")

File: src/respo_alerts/comp/alerts.cljs [hunk 2]
```diff
@@ -104,8 +128,8 @@
          {}
          (input
           {:style (merge ui/input {:width "100%"}),
+           :class-name schema/input-box-name,
            :placeholder "",
-           :autofocus true,
```
Comment (lilydjwg) [inline src/respo_alerts/comp/alerts.cljs:108]: Why not?

Comment (tiye) [reply src/respo_alerts/comp/alerts.cljs:108]: I added `(.select target)` somewhere in the code, which is a replacement for `autofocus`. Plus `autofocus` does not alway work. Not sure why it did work. Maybe caused by timing...

File: src/respo_alerts/comp/alerts.cljs (hunk 1 above)
Comment (tiye) [reply src/respo_alerts/comp/alerts.cljs:30]: > Is there a better way than setTimeout?

No, I can think of one. I have to wait next loop of rendering. Currently there's no API got that in Respo.

> Why not `getElementsByClassName`?

I believe `d.querySelector()` is a more general solution. In this case I don't even worry about performance.

> Is it possible that there are multiple elements with the class?

Unlikely in this app. I'm not going to pop 2 of them.

> What if the caller wants the cancel button to be default? (e.g. "Destroy the world?")

Mouse click. Haven't think about navigating with keyboard yet. Only mouse click by now.

**Review by csvwolf — APPROVED** [review]

File: src/respo_alerts/comp/alerts.cljs (hunk 1 above)
Comment (lilydjwg) [reply src/respo_alerts/comp/alerts.cljs:30]: > Unlikely in this app. I'm not going to pop 2 of them.

Isn't this a library?

Comment (chenyong) [reply src/respo_alerts/comp/alerts.cljs:30]: Yes, it's a library. I was saying I don't pop 2 of them, in the business logics.

Comment (chenyong) [reply src/respo_alerts/comp/alerts.cljs:30]: `alert` `prompt` `confirm` are browser APIs, which are blocking the thread. This library is trying to replacing them with Respo. So they are not intended to have multiple instances in the mean time.

**Pull request merged**